hnn_core/tests/test_network.py

```diff
@@ -50,17 +50,13 @@ def test_network():
     print(network_builder)
     print(network_builder._cells[:2])
 
-    # Assert that proper number of gids are created for Network drives
+    # Assert that proper number/types of gids are created for Network drives
     dns_from_gids = [name for name in net.gid_ranges.keys() if
                      name not in net.cell_types]
-    assert len(dns_from_gids) == len(net.external_drives)
+    assert sorted(dns_from_gids) == sorted(net.external_drives.keys())
     for dn in dns_from_gids:
-        assert dn in net.external_drives.keys()
-        this_src_gids = set([gid for drive_conn in
-                             net.external_drives[dn]['conn'].values() for
-                             gid in drive_conn['src_gids']])  # NB set: globals
-        assert len(net.gid_ranges[dn]) == len(this_src_gids)
-        assert len(net.external_drives[dn]['events']) == 1  # single trial!
+        n_drive_cells = net.external_drives[dn]['n_drive_cells']
```

Comment:
this kind of implicit test is fine for now and I know this is not your code but I guess it would have been better to check against what you fed into `net.add_xxx_drive`. That is by explicitly making a call to `net.add_xxx_drive` rather than relying on `add_drives_from_params` Right now, this is not testing for any inaccuracies that may arise from the transcription of `net.external_drives[dn]['n_drive_cells']`

Reply:
Good point. Ideally we'd add an explicit test for `add_drive_from_params=True` in `test_drives.py` and then deprecate dependency on `add_drive_from_params` in `test_network.py`. I'll start an issue for this.

Reply:
See #394.